api-admin/src/main/java/bio/terra/pearl/api/admin/service/study/StudyExtService.java

```diff
@@ -137,4 +138,15 @@ private StudyEnvironment makeEmptyEnvironment(EnvironmentName envName, boolean i
             .build();
     return studyEnv;
   }
+
+  @EnforcePortalStudyPermission(permission = "study_settings_edit")
+  public Study update(PortalStudyAuthContext authContext, Study study) {
+    study.setId(authContext.getPortalStudy().getStudyId());
+    if (!study.getShortcode().equals(authContext.getStudyShortcode())
+        && !authContext.getOperator().isSuperuser()) {
+      throw new PermissionDeniedException("Study shortcode cannot be changed");
+    }
```

Comment:
I like how you explicitly fail if they try to set the shortcode, instead of just ignoring it

I know that Study only has two fields on it, but I'd still rather be explicit about what we can set via the API for safety.   So I'd lean twoards something like:
```
  Study study = studyService.find(authContext.getPortalStudy().getStudyId()).orElseThrow()
 if (!study.getShortcode().equals(studyUpdate.getStudyShortcode())
        && !authContext.getOperator().isSuperuser()) {
      throw new PermissionDeniedException("Study shortcode cannot be changed");
    }  
   study.setName(studyUpdate.getName());
   study.setShortcode(studyUpdate.getName());
 ```